tests/Unit/AddContactTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+use Inertia\Testing\AssertableInertia as Assert;
+
+class AddContactTest extends \Tests\TestCase
+{
+    use \Illuminate\Foundation\Testing\DatabaseMigrations;
+
+    public function test_add_contact_with_first_and_last_name()
+    {
+        $this->followRedirects = true;
+
+        $data = ['first_name' => 'Bob', 'last_name' => 'Smith'];
+
+        $this->assertCount(0, \App\Models\Contact::query()->where($data)->get());
+
+        $response = $this->postJson('/contact', $data);
+
+        $response->assertInertia(fn(Assert $page) => $page->component('Contact/Index'));
+
+        $this->assertGreaterThan(0, \App\Models\Contact::query()->where($data)->count());
+
+        $this->assertSame(
+            $data,
+            \App\Models\Contact::query()
+                ->firstWhere($data)
+                ->only('first_name', 'last_name')
+        );
+    }
+
+    public function test_add_contact_with_all_details()
+    {
+        $this->followRedirects = true;
+
+        $data = [
+            'first_name' => 'Barry',
+            'last_name' => 'Scott',
+            'email' => 'barry.scott@example.com',
+            'date_of_birth' => \Carbon\Carbon::parse('1977-02-03')
+        ];
+
+        $this->assertCount(0, \App\Models\Contact::query()->where($data)->get());
+
+        $response = $this->postJson('/contact', $data);
+
+        $response->assertInertia(fn(Assert $page) => $page->component('Contact/Index'));
+
+        $this->assertGreaterThan(0, \App\Models\Contact::query()->where($data)->count());
+
+        $contact = \App\Models\Contact::query()
+            ->firstWhere($data);
+
+        $this->assertSame(
+            collect($data)->only('first_name', 'last_name', 'email')->toArray(),
+            $contact->only('first_name', 'last_name', 'email')
+        );
+
+        $this->assertNotSame($data['date_of_birth'], $contact->only('date_of_birth'));
+
+        $contactDateOfBirthArray = $contact->only('date_of_birth');
+
+        $this->assertSame($data['date_of_birth']->toDateString(), $contactDateOfBirthArray['date_of_birth']->toDateString());
+    }
+
+    public function test_add_contact_with_no_first_name()
+    {
+        $this->followRedirects = true;
+
+        $data = ['first_name' => null, 'last_name' => 'Smith'];
+
+        $this->assertCount(0, \App\Models\Contact::query()->where($data)->get());
+
+        $response = $this->postJson('/contact', $data);
+
+        $response->assertJsonValidationErrors(['first_name']);
+
+        $this->assertCount(0, \App\Models\Contact::query()->where($data)->get());
+    }
+
+    public function test_add_contact_with_no_last_name()
+    {
+        $this->followRedirects = true;
+
+        $data = ['first_name' => 'Bob', 'last_name' => null];
+
+        $this->assertCount(0, \App\Models\Contact::query()->where($data)->get());
+
+        $response = $this->postJson('/contact', $data);
+
+        $response->assertJsonValidationErrors(['last_name']);
```

Comment:
try not to use non unit test assertions;

If the response error code is 400 or 404 or 500 (uncaught exception?) you can assertSame (400, $response->getResponseCode())

If a api end point errors for what ever reason it should return a standard expected response for example:

Your Json fails.. try{} catch..

return new Response(400, json_decode('Json blew up'));

self::assertSame(400, $response->getResponseCode());
self::assertSame('Json blew up', $response->getBody()->toString());